### DISCUSSION TITLE
How I can start the simulation from the state where it completed previously?

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#270] [Q&A] [answered]
Hi Dr. Cui,

Is there any way to start the simulation from the point where it completed previously?

E.g.
1. I have started simulation **through a script** for **50 seconds** and it is running continuously at the back end.
2. I have dynamically updated value of `tm0 `of `GENROU_2` at 20 seconds using pert file.
3. I have dynamically updated value of `tm0 `of `GENROU_3` at 30 seconds using pert file.
4. Simulation is completed at 50 seconds.

My query is: 
Now, I have started the simulation again for 50 seconds and I want it to continue from the state where the last simulation ended up.

Regards,
A Singh

**Comment by cuihantao** [comment]
How is is different from example 8 linked here: https://github.com/cuihantao/andes/blob/master/examples/8.%20change-setpoints.ipynb?

**Reply by TheASingh** [reply]
Here I'm running it through a **python script**. In that script, I have added the **code** for importing the ANDES package, loading the system and running the TDS.

I'm running this python script from Jupyter notebook. 
When the simulation is completed, to run the simulation again, I have to re-run the python script, and then new system is loaded.

I want to store the state of previously loaded system. So that I can continue where I left.

**Comment by cuihantao** [comment, marked as answer]
You can turn that script into a Jupyter notebook, which gives you a persistent session. When done, you can set a new end time and to pick up from where it ended, just like shown in the example.

Regards,
Hantao Cui
